File: RetailChainUi/src/services/auth.service.js
import { axiosPrivate, axiosPublic } from './api/axiosClient';

const login = async (email, password) => {
    return await axiosPublic.post('/auth/login', { email, password });
};

const logout = async () => {
    return await axiosPrivate.post('/auth/logout');
};

const getCurrentUser = async () => {
    return await axiosPrivate.get('/user/me');
};

const changePassword = async (data) => {
    return await axiosPrivate.post('/auth/change-password', data);
};

const forgotPassword = async (email) => {
    return await axiosPublic.post(`/auth/forgot-password?email=${email}`);
};

const confirmPassword = async (data) => {
    return await axiosPublic.post('/auth/confirm-password', data);
};

const firstTimeChangePassword = async (data, token) => {
    return await axiosPublic.post('/auth/first-change-password', data, {
        headers: {
            'Authorization': `Bearer ${token}`
        }
    });
};

const verifyOtp = async (email, otp) => {
    return await axiosPublic.post(`/auth/verify-otp?email=${email}&otp=${otp}`);
};

const authService = {
    login,
    logout,
    getCurrentUser,
    changePassword,
    forgotPassword,
    confirmPassword,
    verifyOtp,
    firstTimeChangePassword
};

export default authService;

File: RetailChainService/src/main/java/com/sba301/retailmanagement/service/AuthService.java
package com.sba301.retailmanagement.service;

import com.sba301.retailmanagement.dto.request.ChangePassWordRequest;
import com.sba301.retailmanagement.dto.request.ConfirmPasswordRequest;
import com.sba301.retailmanagement.dto.request.LoginRequest;
import com.sba301.retailmanagement.dto.request.RefreshTokenRequest;
import com.sba301.retailmanagement.dto.request.RegisterRequest;
import com.sba301.retailmanagement.dto.response.AuthResponse;

public interface AuthService {
    AuthResponse login(LoginRequest request);

    AuthResponse register(RegisterRequest request);

    AuthResponse refreshToken(RefreshTokenRequest request);

    void logout(Long userId);

    AuthResponse changePassword(ChangePassWordRequest request, String token);

    void forgotPassWord(String email);

    void confirmPassWord(ConfirmPasswordRequest request);

    boolean verifyOtp(String email, String otp);
}
